Wikipedia/Code/WKScriptMessage+WMFScriptMessage.swift

```diff
@@ -0,0 +1,86 @@
+
+@objc public enum WMFWKScriptMessage: Int {
+    case unknown
+    case consoleMessage
+    case clickLink
+    case clickImage
+    case clickReference
+    case clickEdit
+    case nonAnchorTouchEndedWithoutDragging
+    case lateJavascriptTransform
+    case articleState
+    case findInPageMatchesFound
+    case footerReadMoreSaveClicked
+    case footerReadMoreTitlesShown
+    case footerMenuItemClicked
+    case footerLegalLicenseLinkClicked
+}
+
+extension WKScriptMessage {
+
+    public class func wmf_typeForMessageName(_ name: String) -> WMFWKScriptMessage {
+        switch name {
+        case "nonAnchorTouchEndedWithoutDragging":
+            return .nonAnchorTouchEndedWithoutDragging
+        case "linkClicked":
+            return .clickLink
+        case "imageClicked":
+            return .clickImage
+        case "referenceClicked":
+            return .clickReference
+        case "editClicked":
+            return .clickEdit
+        case "lateJavascriptTransform":
+            return .lateJavascriptTransform
+        case "articleState":
+            return .articleState
+        case "sendJavascriptConsoleLogMessageToXcodeConsole":
+            return .consoleMessage
```

Comment:
Its unfortunate that not all of these match, but I guess it's not too hard to maintain the list